packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -696,217 +702,231 @@ fn process_fn_items(
+                                    contract_id: ContractId::from(<[u8; 32]>::from(id)),
+                                    error_val: u64::from(ra & 0xF)
+                                }
+                            );
+                            decoder.decode_type(ty_id, data)?;
+                        }
+                        fuel::Receipt::Mint { sub_id, contract_id, val, pc, is } => {
+                            let ty_id = Mint::type_id();
+                            let data = serialize(
+                                &Mint {
+                                    sub_id: AssetId::from(<[u8; 32]>::from(sub_id)),
+                                    contract_id: ContractId::from(<[u8; 32]>::from(contract_id)),
+                                    val,
+                                    pc,
+                                    is
+                                }
+                            );
+                            decoder.decode_type(ty_id, data)?;
+                        }
+                        fuel::Receipt::Burn { sub_id, contract_id, val, pc, is } => {
+                            let ty_id = Burn::type_id();
+                            let data = serialize(
+                                &Burn {
+                                    sub_id: AssetId::from(<[u8; 32]>::from(sub_id)),
+                                    contract_id: ContractId::from(<[u8; 32]>::from(contract_id)),
+                                    val,
+                                    pc,
+                                    is
+                                }
+                            );
+                            decoder.decode_type(ty_id, data)?;
+                        }
+                        _ => {
+                            info!("This type is not handled yet. (>'.')>");
+                        }
                     }
                 }
 
-                decoder.dispatch(block.header.height);
+                Ok(())
+            };
+
+            let mut process_block = |block: BlockData| -> anyhow::Result<()> {
+                #start_block
+
+                let mut decoder = Decoders::default();
+
+                let ty_id = BlockData::type_id();
+                let data = serialize(&block);
+                decoder.decode_type(ty_id, data)?;
+
+                for tx in block.transactions {
+                    let tx_id = tx.id;
+                    process_transaction(&mut decoder, tx).with_context(|| format!("Failed processing Transaction {:?}", tx_id))?
+                }
+
+                decoder.dispatch()?;
 
                 let metadata = IndexMetadataEntity::new(block.time as u64, block.header.height, block.id);
                 metadata.save();
+                Ok(())
+            };
+
+            for block in blocks {
+                let block_height = block.header.height;
+                process_block(block).with_context(|| format!("Failed processing Block #{}", block_height))?;
```

Comment:
Shouldn't this come before we save the metadata? Otherwise, we have no guarantee that we actually processed a block prior to incrementing the metadata table.

Reply:
Ah yes, you're completely right. That's what I get for using the side-by-side diff view.